pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -158,7 +160,24 @@ func (r *ReconcileEgressGateway) Reconcile(ctx context.Context, request reconcil
 	}
 
 	// If there are no Egress Gateway resources, return.
+	ch := utils.NewComponentHandler(log, r.client, r.scheme, nil)
 	if len(egws) == 0 {
+		objects := []client.Object{}
+		// If PSP is enabled, remove the Pod Security policy.
+		if r.usePSP {
+			psp := egressgateway.PodSecurityPolicy()
+			objects = append(objects, psp)
+		}
+		// If provider is openshift, remove the SCC
+		if r.provider == operatorv1.ProviderOpenShift {
+			scc := egressgateway.SecurityContextConstraints()
+			objects = append(objects, scc)
+		}
+
+		err := ch.CreateOrUpdateOrDelete(ctx, render.NewPassthrough(true, objects...), r.status)
+		if err != nil {
+			reqLogger.Error(err, "error deleting cluster scoped resources")
```

Comment:
I think we should return an error if we encounter an error here - as it stands, we'll return `nil`